lib/apiservers/engine/backends/convert/state.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/go-units"
+
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// State will create and return a docker ContainerState object
+// from the passed vic ContainerInfo object
+func State(info *models.ContainerInfo) *types.ContainerState {
+	defer trace.End(trace.Begin(""))
+
+	// ensure we have the data we need
+	if info == nil || info.ProcessConfig == nil || info.ContainerConfig == nil {
+		return nil
```

Comment:
I don't know -- I typically subscribe to no data versus partial or incorrect data.  I'll open an issue to investigate, but we won't hold up this PR.